backend/src/scripts/db-scripts.js

```diff
@@ -0,0 +1,106 @@
+const mongoose = require('mongoose');
+const Course = require('../models/course');
+
+
+// duplicate course documents with new ObjectId 
+const duplicateCourses = async (courses, numDup) => {
+    for (let course of courses) {
+        console.log(`duplicating ${course.dept} ${course.num}: ${course.title}...`);
+        for (let i = 0; i < numDup; i++) {
+            course._id = mongoose.Types.ObjectId();
+            course.isNew = true;
+            await course.save()
+        }
+    }
+}
+
+const removePrevQuartersField = async (courses) => {
+    for (let course of courses) {
+        console.log(`removing prevQuarters in ${course.dept} ${course.num}: ${course.title}`);
+        course.prevQuarters = undefined;
+        await course.save();
+    }
+}
+
+
+
+// RUN WHEN COURSES COLLECTION IS INITIALIZED
+const duplicateNecessaryCourses = async () => {
+    let specialTopicCourses = await Course.find({
+        title: { $regex: 'special topic', $options: 'i' }
+    })
+
+    let undergradResearchCreditCourses = await Course.find({
+        $or: [{ title: { $regex: 'independent', $options: 'i' } }, { title: { $regex: 'individual', $options: 'i' } }, { title: { $regex: 'research', $options: 'i' } }],
+        num: { $regex: '199', $options: 'i' },
+    })
+
+    let gradResearchCreditCourses299 = await Course.find({
+        $or: [{ title: { $regex: 'independent', $options: 'i' } }, { title: { $regex: 'individual', $options: 'i' } }, { title: { $regex: 'research', $options: 'i' } }],
+        num: { $regex: '299', $options: 'i' }
+    })
+
+    let gradResearchCreditCourses298 = await Course.find({
+        $or: [{ title: { $regex: 'independent', $options: 'i' } }, { title: { $regex: 'individual', $options: 'i' } }, { title: { $regex: 'research', $options: 'i' } }],
+        num: { $regex: '298', $options: 'i' }
+    })
+
+    let gradResearchCreditCourses290 = await Course.find({
+        $or: [{ title: { $regex: 'independent', $options: 'i' } }, { title: { $regex: 'individual', $options: 'i' } }, { title: { $regex: 'research', $options: 'i' } }],
+        num: { $regex: '290', $options: 'i' }
+    })
+
```

Comment:
Use Promise.all for these 5 duplicateCourses.
Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
@anthopark another good reference here: https://stackoverflow.com/questions/45285129/any-difference-between-await-promise-all-and-multiple-await

Reply:
@emmohac Thanks for the suggestions! I haven't seen Promise.all(). I'll look into this later.

Reply:
@anthopark sure

Reply:
@emmohac can u explain why u suggested Promise.all() here? is it because I can asynchronously perform five Course.find()?

Reply:
@anthopark I think this one would explain better than I do with actual data: https://dev.to/micahriggan/serial-promises-vs-parallel-promises-1ejc

Reply:
@emmohac gotcha, I'll use update this later when I get to touch this function. for now, let's resolve it